### ISSUE TITLE
Add support for Graviton2 - AL2_ARM_64 ami_type for EKS managed node groups

### ISSUE THREAD

**Issue by casey-robertson** [terraform-providers/terraform-provider-aws#14743]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support to EKS : aws_eks_node_group to support the AL2_ARM_64 ami_type (ARM-based Graviton2 processors) for EKS managed node groups using M6 instances

### New or Affected Resource(s)

* aws_eks_node_group / ami_type

### Potential Terraform Configuration

```hcl
resource "aws_eks_node_group" "example" {
  cluster_name    = aws_eks_cluster.example.name
  node_group_name = "example"
  node_role_arn   = aws_iam_role.example.arn
  subnet_ids      = aws_subnet.example[*].id

  scaling_config {
    desired_size = 1
    max_size     = 1
    min_size     = 1
  }
}
  ami_type       = AL2_ARM_64
  disk_size      = 80
  instance_types = m6g.large
```

### References

https://aws.amazon.com/blogs/containers/eks-on-graviton-generally-available/
https://docs.aws.amazon.com/eks/latest/APIReference/API_Nodegroup.html

**Comment by casey-robertson** [comment]
Looks like this PR has addressed the request:  https://github.com/terraform-providers/terraform-provider-aws/pull/14729

**Comment by ewbankkit** [comment]
@casey-robertson Thanks for raising this issue.
Yes, #14729, scheduled to be released with v3.3.0 of the Terraform AWS Provider, addresses this.
